### PR TITLE
merge: combine entries together when Batch Headers match

### PR DESCRIPTION
**Pull request by adamdecaf**
Issue: https://github.com/moov-io/ach/issues/927

### CODE REVIEW

File: batchHeader.go [hunk 1]
```diff
@@ -213,6 +211,30 @@ func (bh *BatchHeader) String() string {
...
+	}
+	if bh.StandardEntryClassCode != other.StandardEntryClassCode {
+		return false
+	}
+	if bh.EffectiveEntryDate != other.EffectiveEntryDate {
+		return false
+	}
+	if bh.ODFIIdentification != other.ODFIIdentification {
+		return false
+	}
```
Comment (vxio) [inline batchHeader.go:234]:
```suggestion
	if bh.ServiceClassCode != other.ServiceClassCode &&
	!strings.EqualFold(bh.CompanyName, other.CompanyName) &&
	bh.CompanyIdentification != other.CompanyIdentification &&
	bh.StandardEntryClassCode != other.StandardEntryClassCode &&
        bh.EffectiveEntryDate != other.EffectiveEntryDate &&
	bh.ODFIIdentification != other.ODFIIdentification {
		return false
	}
```

File: merge.go [hunk 2]
```diff
@@ -151,13 +163,27 @@ func (fs *mergableFiles) findOutfile(f *File) *File {
...
 }
 
+func mergeEntries(b1, b2 Batcher) Batcher {
+	b, _ := NewBatch(b1.GetHeader())
+	entries := sortEntriesByTraceNumber(append(b1.GetEntries(), b2.GetEntries()...))
+	for i := range entries {
+		b.AddEntry(entries[i])
+	}
+	b.SetControl(b1.GetControl())
+	b.Create()
```
Comment (vxio) [inline merge.go]: This should never err right?

**Review by vxio — DISMISSED** [review]
Had minor comments, but lgtm

File: batchHeader.go (hunk 1 above)
Comment (adamdecaf) [reply batchHeader.go:234]: I was debating this. Putting it all into one `if` seems a bit messy. We don't really do that elsewhere in this project.

Comment (vxio) [reply batchHeader.go:234]: I'm fine with leaving as is

File: merge.go (hunk 2 above)
Comment (adamdecaf) [reply merge.go]: Correct. We can check that error - probably should.

**Review by vxio — APPROVED** [review]

**Pull request merged**